### DISCUSSION TITLE
how to get to configuration without printer attached?

### DISCUSSION THREAD

**Discussion by tfaz099** [UnchartedBull/OctoDash#2602] [Q&A]
Once you go through the first config, how do i get back to the configuration settings if it is not connected to a computer.  i.e. Octodash shows on screen that it is sleeping, is there a way to get to a real menu?

thank you

**Comment by jneilliii** [comment]
What setting are you wanting to change? Some of them can be changed with my Octodash Companion plugin. Otherwise, you could SSH to the pi and edit the `/home/pi/.config/octodash/config.json` file directly with nano.

**Comment by tfaz099** [comment]
Thank you
I am looking to be able to switch connections to different Octoprint instances
That menu where I select the printer and then seek the API key.
I see that part within config.json but generating that key automatically with the octoprint request is real easy.  So I sit on the Shhhh octodash is sleeping screen but cannot get to settings ☹


“Once you choose hope, anything’s possible”


On 1/20/22, 9:03 PM, "jneilliii" ***@***.******@***.***>> wrote:


What setting are you wanting to change? Some of them can be changed with my Octodash Companion plugin. Otherwise, you could SSH to the pi and edit the /home/pi/.config/octodash/config.json file directly with nano.

—
Reply to this email directly, view it on GitHub<https://github.com/UnchartedBull/OctoDash/discussions/2602#discussioncomment-2011352>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ATDLIPZ4UNSBC7L7KMI37WTUXC5IVANCNFSM5MOJZ6CA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Reply by jneilliii** [reply]
Oh, I recently posted about setting this up. Maybe this will help you...

https://github.com/UnchartedBull/OctoDash/issues/1148#issuecomment-1007902993

**Comment by tfaz099** [comment]
Thank you again,
Once I tried this and ran this command
cp /home/pi/.config/octodash/config.json /home/pi/.config/octodash/config2.json -f && cp /home/pi/.config/octodash/config1.json /home/pi/.config/octodash/config.json -f && sudo service ***@***.*** restart

it rebooted and halted with a java error

I copied the backup config.json over and it rebooted ok.

I am presuming that the custom actions switch the environment  but those last two entries are unclear what I do with them.  And the fact that it did not work when I rebooted after that copy and service restart makes me a little leary

Im pretty good with some code and compiling etc… but trying not to brick my pi lol


“Once you choose hope, anything’s possible”


On 1/20/22, 9:33 PM, "jneilliii" ***@***.******@***.***>> wrote:


Oh, I recently posted about setting this up. Maybe this will help you...

#1148 (comment)<https://github.com/UnchartedBull/OctoDash/issues/1148#issuecomment-1007902993>

—
Reply to this email directly, view it on GitHub<https://github.com/UnchartedBull/OctoDash/discussions/2602#discussioncomment-2011437>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ATDLIP5NWGNVY52YRC62Q2LUXDAYZANCNFSM5MOJZ6CA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Reply by jneilliii** [reply]
The first half is creating the additional config.json for the second instance, which you then update with new api key, name, e.t.c. the second half is settings for the gcode system commands plugin. If it broke after the first half of configuration before even getting to the second half then you probably messed up the json formatting of the file.

**Comment by tfaz099** [comment]
Odd. I pasted the snippets from your email in the same exact way as the first custom action in both config 1 and 2
Json files



“Don’t adapt to the energy in the room, influence it” 

On Jan 21, 2022, at 8:47 AM, jneilliii ***@***.***> wrote:

﻿
The first half is creating the additional config.json for the second instance, which you then update with new api key, name, e.t.c. the second half is settings for the gcode system commands plugin. If it broke after the first half of configuration before even getting to the second half then you probably messed up the json formatting of the file.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by jneilliii** [reply]
so quick question. is the custom action exactly as you wrote it in your comment, like this?
```
cp /home/pi/.config/octodash/config.json /home/pi/.config/octodash/config2.json -f && cp /home/pi/.config/octodash/config1.json /home/pi/.config/octodash/config.json -f && sudo service ***@***.*** restart
```
It should be one single line and shouldn't have astericks in it, so I think that might be GitHub stripping out what it thinks is an email address. Maybe a screenshot will help...although it gets cut off.

![image](https://user-images.githubusercontent.com/5249455/150561256-e89e844a-709a-4eb0-91c7-3f2a8821990a.png)

if you are getting error, sharing the error would be helpful.

**Comment by tfaz099** [comment]
I will try it again over the weekend (maybe tonight) and will let you know. I tried that from an ssh shell to test it. Never put it into the gcode sections yet



“Don’t adapt to the energy in the room, influence it” 

On Jan 21, 2022, at 11:14 AM, jneilliii ***@***.***> wrote:

﻿
so quick question. is the custom action exactly as you wrote it in your comment, like this?

cp /home/pi/.config/octodash/config.json /home/pi/.config/octodash/config2.json -f && cp /home/pi/.config/octodash/config1.json /home/pi/.config/octodash/config.json -f && sudo service ***@***.*** restart
It should be one single line and shouldn't have astericks in it, so I think that might be GitHub stripping out what it thinks is an email address. Maybe a screenshot will help...although it gets cut off.



if you are getting error, sharing the error would be helpful.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by tfaz099** [comment]
This is my config 2 file:

cat config2.json

{

      "config": {

            "octoprint": {

                  "url": "http://octopi2.local:80/",

                  "accessToken": "82A04B2106FA4350A972BA6A8CBF09B1"

            },

            "printer": {

                  "name": "PortiPi",

                  "xySpeed": 150,

                  "zSpeed": 5,

                  "disableExtruderGCode": "M18 E",

                  "zBabystepGCode": "M290 Z",

                  "defaultTemperatureFanSpeed": {

                        "hotend": 200,

                        "heatbed": 60,

                        "fan": 100

                  }

            },

            "filament": {

                  "thickness": 1.75,

                  "density": 1.25,

                  "feedLength": 80,

                  "feedSpeed": 20,

                  "feedSpeedSlow": 3,

                  "purgeDistance": 30,

                  "useM600": false

            },

            "plugins": {

                  "displayLayerProgress": {

                        "enabled": true

                  },

                  "enclosure": {

                        "enabled": false,

                        "ambientSensorID": null,

                        "filament1SensorID": null,

                        "filament2SensorID": null

                  },

                  "filamentManager": {

                        "enabled": false

                  },

                  "preheatButton": {

                        "enabled": true

                  },

                  "printTimeGenius": {

                        "enabled": true

                  },

                  "psuControl": {

                        "enabled": false,

                        "turnOnPSUWhenExitingSleep": false

                  },

                  "tpLinkSmartPlug": {

                        "enabled": false,

                        "smartPlugIP": "127.0.0.1"

                  },

                  "tasmota": {

                        "enabled": false,

                        "ip": "127.0.0.1",

                        "index": null

                  },

                  "tasmotaMqtt": {

                        "enabled": false,

                        "topic": "topic",

                        "relayNumber": null

                  }

            },

            "octodash": {

                  "customActions": [

                                {

                                        "color": "#a93e26",

                                        "command": "OCTO1",

                                        "confirm": true,

                                        "exit": false,

                                        "icon": "print"

                                }

                        {

                              "icon": "home",

                              "command": "G28",

                              "color": "#dcdde1",

                              "confirm": false,

                              "exit": true

                        },

                        {

                              "icon": "ruler-vertical",

                              "command": "G29",

                              "color": "#4bae50",

                              "confirm": false,

                              "exit": true

                        },

                        {

                              "icon": "fire-alt",

                              "command": "M140 S50; M104 S185",

                              "color": "#e1b12c",

                              "confirm": false,

                              "exit": true

                        },

                        {

                              "icon": "snowflake",

                              "command": "M140 S0; M104 S0",

                              "color": "#0097e6",

                              "confirm": false,

                              "exit": true

                        },

                        {

                              "icon": "redo-alt",

                              "command": "[!RELOAD]",

                              "color": "#7f8fa6",

                              "confirm": true,

                              "exit": false

                        },

                        {

                              "icon": "power-off",

                              "command": "[!SHUTDOWN]",

                              "color": "#e84118",

                              "confirm": true,

                              "exit": false

                        }

                  ],

                  "fileSorting": {

                        "attribute": "name",

                        "order": "asc"

                  },

                  "invertAxisControl": {

                        "x": false,

                        "y": false,

                        "z": false

                  },

                  "pollingInterval": 2000,

                  "touchscreen": true,

                  "turnScreenOffWhileSleeping": true,

                  "turnOnPrinterWhenExitingSleep": false,

                  "preferPreviewWhilePrinting": false,

                  "previewProgressCircle": false,

                  "screenSleepCommand": "xset dpms force standby",

                  "screenWakeupCommand": "xset s off && xset -dpms && xset s noblank"

            }

      }

}



“Once you choose hope, anything’s possible”


On 1/21/22, 11:14 AM, "jneilliii" ***@***.******@***.***>> wrote:


so quick question. is the custom action exactly as you wrote it in your comment, like this?

cp /home/pi/.config/octodash/config.json /home/pi/.config/octodash/config2.json -f && cp /home/pi/.config/octodash/config1.json /home/pi/.config/octodash/config.json -f && sudo service ***@***.*** restart

It should be one single line and shouldn't have astericks in it, so I think that might be GitHub stripping out what it thinks is an email address. Maybe a screenshot will help...although it gets cut off.

[Image removed by sender. image]<https://user-images.githubusercontent.com/5249455/150561256-e89e844a-709a-4eb0-91c7-3f2a8821990a.png>

if you are getting error, sharing the error would be helpful.

—
Reply to this email directly, view it on GitHub<https://github.com/UnchartedBull/OctoDash/discussions/2602#discussioncomment-2016677>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ATDLIP6SSJL52DJDKS2VL53UXGA5TANCNFSM5MOJZ6CA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
